File: ABB.Web/bin/Release/Modules/Shared/Components/PesertaApprovalSirama/pesertaSirama.js
$(document).ready(function () {
    btnPreviousPesertaSirama();
    btnNextPesertaSirama();
});

function btnPreviousPesertaSirama(){
    $('#btn-previous-pesertaSirama').click(function () {
        $("#pesertaTab").getKendoTabStrip().select(0);
    });
}

function btnNextPesertaSirama(){
    $('#btn-next-pesertaSirama').click(function () {
        $("#pesertaTab").getKendoTabStrip().select(2);
    });
}

function onChangeMasaPinjaman(e){
    calculateMasaPinjaman();
    
    var id = e.sender.element[0].id;
    
    if(id === "tgl_akh")
        generateRateSirama();
}

function generateRateSirama(){
    var data = {
        kd_cb: $("#kd_cb").val(),
        kd_product: $("#kd_product").val(),
        no_sppa: $("#no_sppa").val(),
        kd_rk: $("#kd_rk").val(),
        kd_thn: $("#kd_thn").val(),
        no_updt: $("#no_updt").val(),
        nilai_ptg: $("#nilai_ptg").getKendoNumericTextBox().value(),
        tgl_pk: $("#tgl_pk").getKendoDatePicker().value(),
        tgl_awal: $("#tgl_mulai").getKendoDatePicker().value(),
        tgl_akhir: $("#tgl_akh").getKendoDatePicker().value()
    }
    
    var jsonData = JSON.stringify(data);
    
    ajaxPost("/Peserta/GenerateRateSirama", jsonData, (result) => {
        $("#jns_program").getKendoDropDownList().value(result.Model.jns_program);
        $("#kd_kategori").getKendoDropDownList().value(result.Model.kd_kategori);
        $("#stn_rate").getKendoDropDownList().value(result.Model.stn_rate);
        $("#pst_rate").getKendoNumericTextBox().value(result.Model.pst_rate);
        $("#nilai_prm").getKendoNumericTextBox().value(result.Model.nilai_prm);
        $("#loading_prm").getKendoNumericTextBox().value(result.Model.loading_prm);
        $("#total_prm").getKendoNumericTextBox().value(result.Model.total_prm);
    });
}

File: ABB.Domain/Entities/PesertaSirama.cs
using System;

namespace ABB.Domain.Entities
{
    public class PesertaSirama
    {
        public string kd_cb { get; set; }

        public string kd_product { get; set; }

        public string kd_thn { get; set; }

        public string kd_rk { get; set; }

        public string no_sppa { get; set; }

        public Int16 no_updt { get; set; }

        public string kd_tujuan_pinjmn { get; set; }

        public string jns_pembiayaan { get; set; }

        public string jns_jaminan { get; set; }

        public string kd_mtu { get; set; }

        public decimal nilai_ptg { get; set; }

        public DateTime tgl_pk { get; set; }

        public DateTime tgl_mulai { get; set; }

        public DateTime tgl_akh { get; set; }

        public string jns_program { get; set; }

        public string kd_kategori { get; set; }

        public decimal pst_rate { get; set; }

        public Int16 stn_rate { get; set; }

        public decimal nilai_prm { get; set; }

        public decimal loading_prm { get; set; }

        public decimal total_prm { get; set; }

        public int? periode_asuransi { get; set; }
    }
}